src/parser_config.h

```diff
@@ -78,8 +78,13 @@ typedef struct _parser_config {
      /*
       * Data should be encoded as `latin1` when using python converter
       * (implementing `loadtxt` default Python 2 compatibility mode).
+      * The c byte converter is used when the user requested `dtype="S"`.
+      * In this case we go via `dtype=object`, however, loadtxt allows latin1
+      * while normal object to string casts only accept ASCII, so it ensures
+      * that that the object array already contains bytes and not strings.
       */
      bool python_byte_converters;
+     bool c_byte_converters;
```

Comment:
The fact that I need some C-side support for the python side hack sucks a bit :(.  Another solution would be if we had a "default converter" when there is no actual converter.  But then I still would need to use `lambda x: x` as a converter, which would just slow things down...

So I guess this is simply an annoying artifact of supporting latin1 here but limiting to ascii in the core of NumPy.